pw_node_add_listener will not take pw_proxy* and it is a wrong type

--- a/src/3rdparty/chromium/third_party/webrtc/modules/video_capture/linux/pipewire_session.cc
+++ b/src/3rdparty/chromium/third_party/webrtc/modules/video_capture/linux/pipewire_session.cc
@@ -78,8 +78,11 @@ PipeWireNode::PipeWireNode(PipeWireSession* session,
       unique_id_(rtc::ToString(id)) {
   RTC_LOG(LS_VERBOSE) << "Found Camera: " << display_name_;
 
-  proxy_ = static_cast<pw_proxy*>(pw_registry_bind(
-      session_->pw_registry_, id, PW_TYPE_INTERFACE_Node, PW_VERSION_NODE, 0));
+  void *node = pw_registry_bind(
+      session_->pw_registry_, id, PW_TYPE_INTERFACE_Node, PW_VERSION_NODE, 0);
+
+  proxy_ = static_cast<pw_proxy*>(node);
+  node_ = static_cast<pw_node*>(node);
 
   static const pw_node_events node_events{
       .version = PW_VERSION_NODE_EVENTS,
@@ -87,7 +90,7 @@ PipeWireNode::PipeWireNode(PipeWireSession* session,
       .param = OnNodeParam,
   };
 
-  pw_node_add_listener(proxy_, &node_listener_, &node_events, this);
+  pw_node_add_listener(node_, &node_listener_, &node_events, this);
 }
 
 PipeWireNode::~PipeWireNode() {
@@ -119,7 +122,7 @@ void PipeWireNode::OnNodeInfo(void* data, const pw_node_info* info) {
       uint32_t id = info->params[i].id;
       if (id == SPA_PARAM_EnumFormat &&
           info->params[i].flags & SPA_PARAM_INFO_READ) {
-        pw_node_enum_params(that->proxy_, 0, id, 0, UINT32_MAX, nullptr);
+        pw_node_enum_params(that->node_, 0, id, 0, UINT32_MAX, nullptr);
         break;
       }
     }
--- a/src/3rdparty/chromium/third_party/webrtc/modules/video_capture/linux/pipewire_session.h
+++ b/src/3rdparty/chromium/third_party/webrtc/modules/video_capture/linux/pipewire_session.h
@@ -69,6 +69,7 @@ class PipeWireNode {
   static bool ParseFormat(const spa_pod* param, VideoCaptureCapability* cap);
 
   pw_proxy* proxy_;
+  pw_node* node_;
   spa_hook node_listener_;
   PipeWireSession* session_;
   uint32_t id_;
